include/seqan3/io/alignment_file/output.hpp

```diff
@@ -0,0 +1,940 @@
+                        detail::decays_to_ignore_v<reference_t<id_type>>         &&
+                        detail::decays_to_ignore_v<reference_t<offset_type>>     &&
+                        detail::decays_to_ignore_v<reference_t<ref_seq_type>>    &&
+                        detail::decays_to_ignore_v<reference_t<ref_id_type>>     &&
+                        detail::decays_to_ignore_v<reference_t<ref_offset_type>> &&
+                        detail::decays_to_ignore_v<reference_t<align_type>>      &&
+                        detail::decays_to_ignore_v<reference_t<flag_type>>       &&
+                        detail::decays_to_ignore_v<reference_t<mapq_type>>       &&
+                        detail::decays_to_ignore_v<reference_t<qual_type>>       &&
+                        detail::decays_to_ignore_v<reference_t<seq_qual_type>>   &&
+                        detail::decays_to_ignore_v<reference_t<mate_type>>       &&
+                        detail::decays_to_ignore_v<reference_t<tags_type>>       &&
+                        detail::decays_to_ignore_v<reference_t<evalue_type>>     &&
+                        detail::decays_to_ignore_v<reference_t<bit_score_type>>),
+                      "At least one of the columns must not be set to std::ignore.");
+
+        static_assert(detail::decays_to_ignore_v<reference_t<seq_qual_type>> ||
+                      (detail::decays_to_ignore_v<reference_t<seq_type>> &&
+                       detail::decays_to_ignore_v<reference_t<qual_type>>),
+                      "You may not select field::SEQ_QUAL and either of field::SEQ "
+                      "and field::QUAL at the same time.");
+
+        assert(!format.valueless_by_exception());
+        std::visit([&] (auto & f)
+        {
+            if constexpr (!detail::decays_to_ignore_v<reference_t<seq_qual_type>>)
+            {
+                auto zipped = ranges::view::zip(header,
+                                                seq_qual,
+                                                id,
+                                                offset,
+                                                ref_seq,
+                                                ref_id,
+                                                ref_offset,
+                                                align,
+                                                flag,
+                                                mapq,
+                                                mate,
+                                                tags,
+                                                evalue,
+                                                bit_score);
+
+                for (auto && v : zipped)
+                    f.write(stream,
+                            options,
+                            std::get<0>(v),
+                            std::get<1>(v) | view::convert<typename reference_t<seq_qual_type>::alignment_alphabet_type>,
```

Comment:
Here is a working theory that compiles on godbolt ...
```cpp
template <size_t idx, typename element_t>
auto apply(element_t && elem)
{
    if constexpr (idx == 1)
        return elem + 1;
    else if constexpr (idx == 4)
        return elem + 4;
    else 
        return std::forward<element_t>(elem);
}

template <typename ...args_t, 
          size_t ... is>
void foo_(std::index_sequence<is...>,
          args_t && ... args)
{
    /*f.write(stream, options, */(apply<is>(args),...)/*)*/;
}

template <typename ...args_t>
void foo(args_t && ... args)
{
    foo_(std::make_index_sequence<sizeof...(args_t)>{}, std::forward<args_t>(args)...);
}

int main()
{
    foo(1, 2, 3, 4, 5, 6, 7, 8, 9);
}
```

If we would use this you could simplify the entire code and save a lot of redundancies.

Reply:
Of course adapted to your use case.

Reply:
I would suggest to drop column-wise support for alignment-file!